edgelet/edgelet-core/src/settings.rs

```diff
@@ -146,17 +146,41 @@ impl SymmetricKeyAttestationInfo {
 pub struct X509AttestationInfo {
     #[serde(skip_serializing_if = "Option::is_none")]
     registration_id: Option<String>,
-    identity_cert: PathBuf,
-    identity_pk: PathBuf,
+    #[serde(with = "url_serde")]
+    identity_cert: Url,
+    #[serde(with = "url_serde")]
+    identity_pk: Url,
 }
 
 impl X509AttestationInfo {
-    pub fn identity_cert(&self) -> &Path {
-        &self.identity_cert.as_path()
+    pub fn identity_cert(&self) -> Result<PathBuf, CertificateConfigError> {
```

Comment:
There's no Error struct in this crate already?

Reply:
Moved these to the crate's Error.